crypto/polynomial/polynomial.go

```diff
@@ -87,6 +80,12 @@ func (p *Polynomial) Differentiate(diffTime uint32) *Polynomial {
 		tempCoeff = new(big.Int).Mod(tempCoeff, p.fieldOrder)
 		diffCoeffSlice[i-diffTime] = tempCoeff
 	}
+	if diffTime >= lengthPolyACoeff {
```

Comment:
Add more tests for this case

Reply:
We have a corner test in Line 73 of polynomial_test.